opd_download_page.py

```diff
@@ -129,9 +129,11 @@ def get_years(selectbox_sources, selectbox_states, selectbox_table_types):
 collect_help = "This collects the data from the data source such as a URL and will make it ready for download. This may take some time."
 
 if st.session_state['show_download'] == True:
-    if st.download_button('Download CSV', data=st.session_state['csv_text_output'] , file_name="selected_rows.csv", mime='text/csv'):
+    
+    csv_filename = opd.data.get_csv_filename(selectbox_states, selectbox_sources, selectbox_states, selectbox_table_types, selected_year)
```

Comment:
Function definition for get_csv_filename is:
`def get_csv_filename(state, source_name, agency, table_type, year)`

You are passing selectbox_states as agency. You should be able to pass selected_rows.iloc[0]["Agency"] as the agency.

You should be able to test that filename is correct by doing the following: 
```
import openpolicedata as opd
src = opd.Source(selectbox_sources, selectbox_states)
download_folder = "download_folder_location"
src.load_from_csv(selected_year, output_dir=download_folder, table_type=selectbox_table_types)
```

Reply:
@sowdm I tested with Arizona_Gilbert_CALLS_FOR_SERVICE_2018.csv, Arizona_Gilbert_CALLS_FOR_SERVICE_2023.csv, Virginia_Fairfax_County_ARRESTS_2019.csv, Virginia_Fairfax_County_ARRESTS_2021.csv. Ready for you to review again